apps/desktop/src/lib/pr/prContent.svelte.ts

```diff
@@ -1,114 +1,70 @@
-import { persistWithExpiration, type Persisted } from '@gitbutler/shared/persisted';
 import type { DetailedCommit } from '$lib/vbranches/types';
 
-export function persistedPRBody(projectId: string, seriesName: string): Persisted<string> {
-	return persistWithExpiration('', 'seriesCurrentPRBody_' + projectId + '_' + seriesName, 5);
-}
-
-export function persistedPRTitle(projectId: string, seriesName: string): Persisted<string> {
-	return persistWithExpiration('', 'seriesCurrentPRTitle_' + projectId + '_' + seriesName, 5);
-}
+export class ReactivePRTitle {
+	private title = $state<string>();
 
-export interface PRContent {
-	title: string;
-	body: string;
-	templateBody: string | undefined;
-}
-
-export interface PRContentParams {
-	isDisplay: boolean;
-	projectId: string;
-	seriesName: string;
-	seriesDescription: string | undefined;
-	existingTitle: string | undefined;
-	existingBody: string | undefined;
-	commits: DetailedCommit[];
-}
-
-export default function getPRContent(params: PRContentParams): PRContent {
-	let templateBody = $state<string | undefined>(undefined);
-	const defaultTitle: string = $derived.by(() => {
-		if (params.isDisplay) return params.existingTitle ?? '';
+	constructor(
+		private isDisplay: boolean,
+		private existingTitle: string | undefined,
+		private commits: DetailedCommit[],
+		private branchName: string
+	) {
+		this.title = this.getDefaultTitle();
+	}
 
+	private getDefaultTitle(): string {
+		if (this.isDisplay) return this.existingTitle ?? '';
 		// In case of a single commit, use the commit summary for the title
-		if (params.commits.length === 1) {
-			const commit = params.commits[0];
+		if (this.commits.length === 1) {
+			const commit = this.commits[0];
 			return commit?.descriptionTitle ?? '';
 		}
+		return this.branchName;
+	}
 
-		return params.seriesName;
-	});
+	get current() {
+		return this.title ?? '';
+	}
 
-	const defaultBody: string = $derived.by(() => {
-		if (params.isDisplay) return params.existingBody ?? '';
-		if (params.seriesDescription) return params.seriesDescription;
-		if (templateBody) return templateBody;
+	set(value: string) {
+		this.title = value;
+	}
```

Comment:
ought this be `set current(value: string) {` ?

Reply:
We can do that, but I thought it would be nice to have a setter method that's explicit.

I have no strong feelings against it, though